assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -1,33 +1,68 @@
-import React from "react"
+import React, { ReactElement } from "react"
 import {
   useMinischeduleBlock,
   useMinischeduleRun,
 } from "../../hooks/useMinischedule"
-import { Block, Run } from "../../minischedule"
+import { Block, Break, Piece, Run, Trip } from "../../minischedule"
 import { TripId } from "../../schedule"
 import Loading from "../loading"
 
 export interface Props {
   activeTripId: TripId
 }
-export const MinischeduleRun = ({ activeTripId }: Props) => {
+export const MinischeduleRun = ({ activeTripId }: Props): ReactElement => {
   const run: Run | null | undefined = useMinischeduleRun(activeTripId)
   if (run === undefined) {
     return <Loading />
   } else if (run === null) {
     return <>No run found</>
   } else {
-    return <div>{JSON.stringify(run)}</div>
+    return (
+      <>
+        {run.activities.map((activity) =>
+          isPiece(activity) ? (
+            <Piece piece={activity} key={activity.start.time} />
+          ) : (
+            <Break break={activity} key={activity.startTime} />
+          )
+        )}
+      </>
+    )
   }
 }
 
-export const MinischeduleBlock = ({ activeTripId }: Props) => {
+export const MinischeduleBlock = ({ activeTripId }: Props): ReactElement => {
   const block: Block | null | undefined = useMinischeduleBlock(activeTripId)
   if (block === undefined) {
     return <Loading />
   } else if (block === null) {
     return <>No block found</>
   } else {
-    return <div>{JSON.stringify(block)}</div>
+    return (
+      <>
+        {block.pieces.map((piece) => (
+          <Piece piece={piece} key={piece.start.time} />
+        ))}
+      </>
+    )
   }
 }
+
+const Break = ({ break: breakk }: { break: Break }) => (
```

Comment:
I wouldn't be surprised if these break out into their own files as they get fleshed out, but this is fine for now. (Though as I think about it you might want to think about whether breaking them out sooner than later will help avoid conflicts as you start working on things in parallel…)

Reply:
Yeah. They might eventually want their own unit/snapshot tests too if we start doing interesting logic on them, but not yet.